### PR TITLE
Support endpoint security parameters in api_params.yaml

### PR DESCRIPTION
**Pull request by wasuradananjith**
## Purpose
Fixes: https://github.com/wso2/product-apim-tooling/issues/162

## Goals
Enhance the current capability of defining endpoint security parameters in api.yaml file by providing the capability to override them using api_params.yaml file, if the user wants.

## Approach
This procedure modiied the template of api_params.yaml as shown in the below example. The fields endpointSecured, endpointUTUsername, endpointUTPassword and endpointAuthDigest have been included in it for a particular environment.
```
environments:
  - name: test
    endpoints:
     production:
       url: 'http://dev.doo.com'
       config:
         retryTimeOut: 60
         retryDelay: 70
         factor: 2
     sandbox:
        url: 'http://test.foo.sandbox.com'
     security:
         enabled: true
         type: basic ( or digest )
         username: admin
         password: admin
```

When assigning the security parameters, following procedure has taken place.
- Checked whether endpointSecured field is there in api_params.yaml. If not, the old procedure was taken place by getting the values from api.yaml file.
- If endpointSecured was set (can either be true or false) in api_params.yaml, that value in api.yaml file was overridden and the steps below were followed.
  - Checked whether endpointSecured == true,
    - If endpointSecured == true, checked whether endpointUTUsername and endpointUTPassword was set in api_params.yaml file. If those were not set, show the user an error asking to set those. When those were set, used those values as security parameters (by overriding the values in api.yaml) when importing the API.
    - If endpointAuthDigest was set (can either be true or false), that value in api.yaml was overridden. If not set, used the value in the api.yaml as in default procedure.
  - If endpointSecured == false,
    - Overrided the endpointUTUsername and the endpointUTPassword as empty in api.yaml file so that endpoint security will not be enabled for sure.

## User stories
-  If endpoint security parameters were set in api_params.yaml file, override the values in api.yaml file.
- If endpoint security parameters were  not set in api_params.yaml file, use the values in api.yaml file. (The old procedure)

## Release note
Enhance the current capability of defining endpoint security parameters in api.yaml file by providing the capability to override them using api_params.yaml file, if the user wants.

## Documentation
Documentation changes should be done. 

## Test environment
-  JDK 1.8.0_241
- Ubuntu 18.04.4 LTS

### CODE REVIEW

File: import-export-cli/cmd/importAPI.go [hunk 1]
```diff
@@ -192,6 +198,66 @@ func mergeAPI(apiDirectory string, environmentParams *params.Environment) error
...
+		// Convert the string endpointSecured to boolean
+		boolEndpointSecured, err := strconv.ParseBool(environmentParamsEndpoints.EndpointSecured)
+		if err != nil {
+			return err
+		}
+		if _, err := api.SetP(boolEndpointSecured, "endpointSecured"); err != nil {
+			return err
+		}
+		if boolEndpointSecured {
+			// If endpoint security is enabled, check whether the username and password has set in api_params.yaml
```
Comment (uvindra) [inline import-export-cli/cmd/importAPI.go]: Better to put the logic inside this if condition into a private function to improve readability. This reduces the level of nesting which makes the logic clear and less error prone

**Comment by malinthaprasan** [comment]
```
environments:
  - name: test
    endpoints:
     production:
       url: 'http://dev.doo.com'
       config:
         retryTimeOut: 60
         retryDelay: 70
         factor: 2
     sandbox:
        url: 'http://test.foo.sandbox.com'
     security:
         enabled: true
         type: basic ( or digest )
         username: admin
         password: admin
```

**Comment by wasuradananjith** [comment]
> Shall we use the below format?
> 
> ```
> environments:
>   - name: test
>     endpoints:
>      production:
>        url: 'http://dev.doo.com'
>        config:
>          retryTimeOut: 60
>          retryDelay: 70
>          factor: 2
>      sandbox:
>         url: 'http://test.foo.sandbox.com'
>      security:
>          enabled: true
>          type: basic ( or digest )
>          username: admin
>          password: admin
> ```

environments:
  - name: test
    endpoints:
     production:
       url: 'http://dev.doo.com'
       config:
         retryTimeOut: 60
         retryDelay: 70
         factor: 2
     sandbox:
        url: 'http://test.foo.sandbox.com'
     security:
         enabled: true
         type: basic ( or digest )
         username: admin
         password: admin

**Comment by wasuradananjith** [comment]
> Shall we use the below format?
> 
> ```
> environments:
>   - name: test
>     endpoints:
>      production:
>        url: 'http://dev.doo.com'
>        config:
>          retryTimeOut: 60
>          retryDelay: 70
>          factor: 2
>      sandbox:
>         url: 'http://test.foo.sandbox.com'
>      security:
>          enabled: true
>          type: basic ( or digest )
>          username: admin
>          password: admin
> ```

According to the template above the logic changed as follows.
- Checked whether enabled field under security is there in api_params.yaml. If not, the old procedure was taken place by getting the values from api.yaml file.
- If enabled field was set (can either be true or false) in api_params.yaml, the value endpointSecured in api.yaml file was overridden and the steps below were followed.
  - Checked whether enabled == true,
    - If enabled == true, checked whether username, password and type were set in api_params.yaml file. If those were not set, show the user an error asking to set those. When those were set, used those values as security parameters (by overriding the values in api.yaml) when importing the API. In here if the type is digest, the endpointAuthDigest in api.yaml was set to true, but if the type is basic, the endpointAuthDigest was set to false. If type is neither basic nor digest, an error message was displayed to the user.
  - If enabled == false,
    - Overrided the endpointUTUsername and the endpointUTPassword as empty in api.yaml file so that endpoint security will not be enabled for sure.

**Pull request merged**